src/new/wallet.js

```diff
@@ -259,6 +259,63 @@ class HathorWallet extends EventEmitter {
     this.storage.config.setServerUrl(newServer);
   }
 
+  /**
+   * Set the value of the gap limit for this wallet instance.
+   * @param {number} value The new gap limit value
+   * @returns {Promise<void>}
+   */
+  async setGapLimit(value) {
+    return this.storage.setGapLimit(value);
+  }
+
+  /**
+   * Get the value of the gap limit for this wallet instance.
+   * @returns {Promise<number>}
+   */
+    async getGapLimit() {
+      return this.storage.getGapLimit();
+    }
+
+  /**
+   * Get the access data object from storage.
+   * @returns {Promise<import('../types').IWalletAccessData>}
+   */
+  async getAccessData() {
+    const accessData = await this.storage.getAccessData();
+    if (!accessData) {
+      throw new WalletError('Wallet was not initialized.');
+    }
+    return accessData;
+  }
+
+  /**
+   * Get the configured wallet type.
+   * @returns {Promise<string>} The wallet type
+   */
+  async getWalletType() {
+    const accessData = await this.getAccessData();
+    return accessData.walletType;
+  }
+
+  /**
+   * Get the multisig data object from storage.
+   * Only works if the wallet is a multisig wallet.
+   *
+   * @returns {Promise<import('../types').IMultisigData>}
+   */
+  async getMultisigData() {
+    const accessData = await this.getAccessData();
+    if (accessData.walletType === WalletType.MULTISIG) {
+      if (accessData.multisigData) {
+        return accessData.multisigData;
+      } else {
+        throw new WalletError('Multisig data not found in storage');
+      }
+    } else {
+      throw new WalletError('Wallet is not a multisig wallet.');
```

Comment:
I would suggest flattening this nested `if` clauses in a way that the most expected result is on the last instruction, and the exceptions are within the `if` blocks. ( "Guard Clauses" [refactoring example](https://refactoring.guru/replace-nested-conditional-with-guard-clauses) ).

It would result in something like:
```js
if (accessData.walletType !== WalletType.MULTISIG) {
  throw new WalletError('Wallet is not a multisig wallet.');
}

if (!accessData.multisigData) {
  throw new WalletError('Multisig data not found in storage');
}

return accessData.multisigData;
```

Reply:
Done